### PR TITLE
Allow for mixed case github usernames

### LINKED ISSUE #54: Proposal:  Case insensitive username handling
We should handle usernames as case insensitive from the CUSTOMERS File

```
Affix
affix
aFfIx
```

should all match to affix

gh-push handler.go line 62

```go
for _, customer := range customers {
	if customer == pushEvent.Repository.Owner.Login {
		found = true
	}
}
```

should be

```go
for _, customer := range customers {
	if strings.EqualFold(customer, pushEvent.Repository.Owner.Login) {
		found = true
	}
}
```

### PR DESCRIPTION
**Pull request by burtonr**
## Description
This change allows customers with mixed case GitHub usernames
to build and deploy functions.

Resolves #54 

Signed-off-by: Burton Rheutan <rheutan7@gmail.com>

## How Has This Been Tested?

Created a new test account @BurtonRTestAccnt with mixed case, installed my OpenFaaS-Cloud app there, created a repo for a function, and pushed to master.

Verified by watching the logs that all steps were successful. Completed the verification by loading the dashboard and seeing the function built. Clicking the link calls the function as expected.

Others are able to test this by either building the new function images themselves, or updating the `stack.yml` for the following functions:
* github-push -> `burtonr/github-push:0.5.0-rc1.4`
* git-tar -> `burtonr/of-git-tar:0.7.0-rc1.1`
* list-functions -> `burtonr/list-functions:0.4.2-1`
> apparently, I'm bad at naming conventions... sorry about that!

## Checklist:

I have:

- [ ] updated the documentation if required **not required for this change**
- [X] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [X] signed-off my commits with `git commit -s`
- [ ] added unit tests

> Sorry, I did not add unit tests for this, but I could give it a shot in a follow up PR if that's acceptable

### CODE REVIEW

**Comment by alexellis** [comment]
Quick thought.. Does this cover of-router?

**Comment by burtonr** [comment]
I'm not sure. I couldn't find reference to of-router in any of the docs. 

It looks like it's only used/deployed when using the Kubernetes deployment. I had only setup and run on Swarm. That is something that would need verified though!

I also couldn't find where, or how of-router is called/used. 

Looking through the of-router `main.go` it looks like it just forwards the request to the `<gateway>/function/<username>-<function name>`. If that's the case, it should work fine as now the username is converted to lower case before any of the build or deploy steps happen. It _should_ be fine...

**Comment by alexellis** [comment]
Rater than mutating the data, I think we need to store the original username in its proper casing and at all times convert to lower for comparisons, formatting etc.

**Comment by alexellis** [comment]
of-router also works with swarm, the current live environment is setup with that. You can just use a host files entry to test it.

Imagine if our domain was `*.case-test.team-serverless.xyz`, then each user would have `username.case-test.team-serverless.xyz`. You'd have `burtonr.case-test.team-serverless.xyz`

So if you don't want to setup Route53/Google DNS or similar you can edit your hosts file

```
this.is.the.ip burtonr.case-test.team-serverless.xyz
```

File: git-tar/function/handler.go [hunk 1]
```diff
@@ -26,6 +27,9 @@ func Handle(req []byte) []byte {
 		os.Exit(-1)
 	}
 
+	// force github username to lowercase to prevent issues downstream
+	pushEvent.Repository.Owner.Login = strings.ToLower(pushEvent.Repository.Owner.Login)
```
Comment (alexellis) [inline git-tar/function/handler.go]: Not sure we should mutate the value from GitHub, this also invalidates the SHA signature.

**Comment by burtonr** [comment]
Apologies for the delay! I've been a bit busy over the weekend. 

I understand the concern with modifying the data. I went that route to fix it at the earliest possible place so that all the downstream services (and future services) wouldn't need to worry about case checking.

Give me a couple of days, and I'll update this to take the value as-is, and change it, or invariant-ly check it, through all the different services. I'll also need to re-base my test cluster the recent updates that happened since I started this. I'm hopeful for a Wednesday update.

**Comment by burtonr** [comment]
Alright, I got it this time. This changeset looks a lot better than the first attempt! Thanks @alexellis  for pushing for doing it the right way.

I've updated the services to put the values to lower case where it's necessary, rather than forcing it at the source. This way, the repo owner and repo names are passed around and stored as-is, then converted to lower case only when they are being used for image names and validation. I'm much happier with this.

File: buildshiprun/handler.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ func Handle(req []byte) string {
 		Source: "buildshiprun",
 	}
 
-	serviceValue := fmt.Sprintf("%s-%s", event.Owner, event.Service)
+	serviceValue := strings.ToLower(fmt.Sprintf("%s-%s", event.Owner, event.Service))
```
Comment (alexellis) [inline buildshiprun/handler.go]: We don't need to change the service name, this is unrelated to GitHub usernames.

**Comment by alexellis** [comment]
Hey a rebase is needed now @burtonr 

I'd also like to know that the dashboard still works and of-router

Thanks,

Alex

**Comment by burtonr** [comment]
@alexellis I've done the rebase. This is ready to go now.

I've verified that the dashboard does still function as expected. Using either lower case, or mixed case in the query string brings up the same page, and the "Endpoint" `Link` routes correctly to the function.

For the of-router, I wasn't able to get that functional in my setup, but following the readme, and watching the logs, it was attempting to route to the correct path. I'm confident that of-router will work as expected.

**Comment by alexellis** [comment]
I can still see the username getting mutated in one place. https://github.com/openfaas/openfaas-cloud/pull/95/files#r209429701

**Comment by burtonr** [comment]
@alexellis That must have gotten mixed up in my rebase from earlier. That was not in the function that I had deployed and tested with.

I've updated the commit to remove that data manipulation
